scope/src/doctor/runner.rs

```diff
@@ -603,61 +644,173 @@ mod tests {
 
         let exit_code = run_groups.execute().await?;
         assert!(exit_code.did_succeed);
-
+        assert_eq!(
+            BTreeSet::from_iter(run_groups.all_paths),
+            exit_code.succeeded_groups
+        );
+        assert_eq!(BTreeSet::new(), exit_code.failed_group);
+        assert_eq!(BTreeSet::new(), exit_code.skipped_group);
         Ok(())
     }
 
     #[tokio::test]
     async fn test_execute_dep_fails_wont_run_others() -> Result<()> {
         let group_actions = BTreeMap::from([
             make_group_action(
-                "group_1",
-                make_action_run(ActionRunStatus::CheckFailedFixSucceedVerifyFailed),
+                "fails",
+                make_action_runs(ActionRunStatus::CheckFailedFixSucceedVerifyFailed),
             ),
-            make_group_action("group_2", will_not_run()),
-            make_group_action("group_3", will_not_run()),
+            make_group_action("skipped_1", will_not_run()),
+            make_group_action("skipped_2", will_not_run()),
         ]);
 
         let run_groups = RunGroups {
             group_actions,
             all_paths: vec![
-                "group_1".to_string(),
-                "group_2".to_string(),
-                "group_3".to_string(),
+                "fails".to_string(),
+                "skipped_1".to_string(),
+                "skipped_2".to_string(),
```

Comment:
If possible, please split these test refactorings out into a separate PR. That will help reduce the size of this behavior changing PR, whilst also improving code quality (and helping speed up PR approval).

Reply:
Not a problem. I thought this diff would go away once #183 was merged. Will look into what’s going on here and either update this PR or submit one with just these refactors.

Reply:
A rebase fixed this up.